src/components/result-ui/page-navigation.js

```diff
@@ -0,0 +1,20 @@
+import React, { useContext } from 'react';
+import {LocationContext} from "../../contexts/location";
+
+const PageNavigation = ({links}) => {
+  const location = useContext(LocationContext);
+  const {first, prev, next, last} = links;
+
+  const followLink = link => () => {
+    location.setUrl(link.href);
+  };
+
+  return (<div className="page_navigation">
+    {first && <span title="Go to first page" className="page_navigation__link_arrow" onClick={followLink(first)}>&laquo;</span>}
```

Comment:
This is a very clear and readable way to handle this. Nice job including the text titles too for accessibility!